internal/tasks/manifests.go

```diff
@@ -64,40 +64,31 @@ var outdatedManifestSearchQuery = sqlext.SimplifyWhitespace(`
 //attribute of their constituent images.
 
 // ValidateNextManifest validates manifests that have not been validated for more
-// than 6 hours. At most one manifest is validated per call. If no manifest
-// needs to be validated, sql.ErrNoRows is returned.
-func (j *Janitor) ValidateNextManifest() (returnErr error) {
-	var manifest keppel.Manifest
-
-	defer func() {
-		if returnErr == nil {
-			validateManifestSuccessCounter.Inc()
-		} else if returnErr != sql.ErrNoRows {
-			validateManifestFailedCounter.Inc()
-			if manifest.Digest == "" || manifest.RepositoryID == 0 {
-				returnErr = fmt.Errorf("while validating a manifest: %w", returnErr)
-			} else {
-				returnErr = fmt.Errorf("while validating manifest %s in repo %d: %w", manifest.Digest, manifest.RepositoryID, returnErr)
-			}
-		}
-	}()
-
-	//find manifest: validate once every 24 hours, but recheck after 10 minutes if
-	//validation failed
-	maxSuccessfulValidatedAt := j.timeNow().Add(-24 * time.Hour)
-	maxFailedValidatedAt := j.timeNow().Add(-10 * time.Minute)
-	err := j.db.SelectOne(&manifest, outdatedManifestSearchQuery, maxSuccessfulValidatedAt, maxFailedValidatedAt)
-	if err != nil {
-		if err == sql.ErrNoRows {
-			logg.Debug("no manifests to validate - slowing down...")
-			return sql.ErrNoRows
-		}
-		return err
-	}
+// than 6 hours. At most one manifest is validated per call.
+func (j *Janitor) ValidateManifestJob(registerer prometheus.Registerer) jobloop.Job {
+	return (&jobloop.ProducerConsumerJob[keppel.Manifest]{
+		Metadata: jobloop.JobMetadata{
+			ReadableName: "manifest syncs in replica repos",
+			CounterOpts: prometheus.CounterOpts{
+				Name: "keppel_blob_validations",
+				Help: "Counter for blob validations.",
```

Comment:
There's a copy-paste error here, which now makes me concerned there's going to be more copy-paste errors elsewhere. Please have another look.